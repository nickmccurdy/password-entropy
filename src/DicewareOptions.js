import Options from './Options'
import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'

export default class DicewareOptions extends PureComponent {
  static propTypes = {
    onChange: PropTypes.func.isRequired
  }

  possiblePasswords = () => 7776

  render () {
    return (
      <Options possiblePasswords={this.possiblePasswords} {...this.props}>
        <h3>Diceware</h3>
      </Options>
    )
  }
}
