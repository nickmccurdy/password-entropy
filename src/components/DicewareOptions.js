import Options from './Options'
import React, { Component } from 'react'

export default class DicewareOptions extends Component {
  static propTypes = Options.sharedPropTypes
  static shortName = 'Diceware'

  possiblePasswords = () => 7776

  render () {
    return (
      <Options possiblePasswords={this.possiblePasswords} {...this.props}>
        <h3>Diceware</h3>
      </Options>
    )
  }
}
