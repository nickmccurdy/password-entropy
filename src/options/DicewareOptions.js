import { TextField } from 'material-ui'
import { ImageEdit } from 'material-ui/svg-icons'
import React from 'react'

export const dicewareWords = 7776

export default function DicewareOptions() {
  return (
    <TextField
      disabled
      floatingLabelText={<div><ImageEdit /> Possible Items</div>}
      value={dicewareWords}
    />
  )
}
