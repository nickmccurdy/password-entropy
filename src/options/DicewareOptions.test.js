import DicewareOptions from './DicewareOptions'
import { shallow } from 'enzyme'
import React from 'react'

test('DicewareOptions', () => {
  expect(shallow(<DicewareOptions/>)).toMatchSnapshot()
})
