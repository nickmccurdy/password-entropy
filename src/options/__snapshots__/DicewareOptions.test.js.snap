// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DicewareOptions 1`] = `
<TextField
  disabled={true}
  floatingLabelFixed={false}
  floatingLabelText={
    <div>
      <ImageEdit />
       Possible Items
    </div>
  }
  fullWidth={false}
  multiLine={false}
  rows={1}
  type="text"
  underlineShow={true}
  value={7776}
/>
`;
